src/features/recovery/RecoveryDialogs.tsx

```diff
@@ -0,0 +1,410 @@
+                session={session}
+                onRecoverClick={onRecoverClick}
+                onRemoveClick={onRemoveClick}
+            />
+        ))}
+    </div>
+);
+
+export default () => {
+    const dispatch = useDispatch();
+    const recoveryManager = RecoveryManager.getInstance();
+
+    const pendingRecovery = useSelector(isSessionRecoveryPending);
+    const savePending = useSelector(isSavePending);
+    const [isSessionsListDialogVisible, setIsSessionsListDialogVisible] =
+        useState(false);
+    const [orphanedSessions, setOrphanedSessions] = useState<Session[]>([]);
+    const [isSearching, setIsSearching] = useState(false);
+    const [sessionSearchProgress, setSessionSearchProgress] = useState(0);
+    const [isRecovering, setIsRecovering] = useState(false);
+    const [recoveryProgress, setRecoveryProgress] = useState(0);
+    const [recoveryError, setRecoveryError] = useState<string | null>(null);
+    const [confirmationDialogConfig, setConfirmationDialogConfig] = useState({
+        isVisible: false,
+        title: '',
+        message: '',
+        confirmText: 'Confirm',
+        cancelText: 'Cancel',
+        onConfirm: () => {},
+        onCancel: () => {},
+    });
+
+    const closeConfirmationDialog = () =>
+        setConfirmationDialogConfig(c => ({ ...c, isVisible: false }));
+    const lastProgress = useRef(-1);
+    const { time, reset, pause, start } = useStopwatch({
+        autoStart: true,
+        resolution: 1000,
+    });
+
+    useEffect(() => {
+        if (!isRecovering) {
+            lastProgress.current = -1;
+            pause();
+        } else {
+            start(0);
+        }
+    }, [isRecovering, pause, start]);
+
+    useEffect(() => {
+        if (recoveryProgress < lastProgress.current) {
+            lastProgress.current = recoveryProgress;
+            reset();
+        }
+    }, [recoveryProgress, reset]);
+
+    useEffect(() => {
+        if (pendingRecovery) {
+            dispatch(
+                addConfirmBeforeClose({
+                    id: 'unsavedData',
+                    message:
+                        'There is a session recovery in progress. If you close the application the recovery progress will be lost and the session will remain in the recovery list. Are you sure you want to close?',
+                })
+            );
+        } else {
+            dispatch(clearConfirmBeforeClose('unsavedData'));
+        }
+    }, [dispatch, pendingRecovery]);
+
+    useEffect(() => {
+        if (savePending) {
+            dispatch(
+                addConfirmBeforeClose({
+                    id: 'unsavedData',
+                    message:
+                        'You have unsaved data. If you close the application this data will be lost. Are you sure you want to close?',
+                })
+            );
+        } else {
+            dispatch(clearConfirmBeforeClose('unsavedData'));
+        }
+    }, [dispatch, savePending]);
```

Comment:
This is not needed as when you `dispatch(setSavePending(true));` and `dispatch(setSavePending(false));` the code in SidePanel.tsx will ensue this automatically